product-catalog/test/productcatalog/pages/ProductDetailPageContentTest.java

```diff
@@ -0,0 +1,134 @@
+package productcatalog.pages;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.neovisionaries.i18n.CountryCode;
+import common.categories.CategoryUtils;
+import common.cms.CmsPage;
+import common.pages.*;
+import common.prices.PriceFinder;
+import common.utils.PriceFormatter;
+import common.utils.Translator;
+import io.sphere.sdk.categories.Category;
+import io.sphere.sdk.categories.CategoryTree;
+import io.sphere.sdk.models.Image;
+import io.sphere.sdk.models.ImageDimensions;
+import io.sphere.sdk.products.ProductProjection;
+import io.sphere.sdk.products.ProductVariant;
+import io.sphere.sdk.shippingmethods.ShippingRate;
+import org.javamoney.moneta.Money;
+import org.junit.Test;
+import productcatalog.models.RichShippingRate;
+
+import javax.money.CurrencyUnit;
+import javax.money.Monetary;
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.util.List;
+import java.util.Locale;
+import java.util.Optional;
+
+import static common.JsonUtils.readJsonNodeFromResource;
+import static common.products.ProductUtils.getProductById;
+import static common.products.ProductUtils.getQueryResult;
+import static io.sphere.sdk.json.JsonUtils.readObjectFromResource;
+import static io.sphere.sdk.json.JsonUtils.toJsonNode;
+import static java.util.Arrays.asList;
+import static java.util.Collections.emptyList;
+import static java.util.stream.Collectors.toList;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class ProductDetailPageContentTest {
+    private final CurrencyUnit eur = Monetary.getCurrency("EUR");
+    private final CountryCode de = CountryCode.DE;
+    private final Locale german = Locale.GERMAN;
+    private final ZonedDateTime todayDe = ZonedDateTime.of(LocalDateTime.now(), ZoneId.of("Europe/Berlin"));
+
+    private final Translator translator = Translator.of(german, emptyList(), emptyList());
+    private final PriceFinder priceFinder = PriceFinder.of(eur, de, Optional.empty(), Optional.empty(), todayDe);
+    private final PriceFormatter priceFormatter = PriceFormatter.of(german);
+
+    private final CategoryTree categories = CategoryTree.of(CategoryUtils.getQueryResult("categoryQueryResult.json").getResults());
+    private final List<ProductProjection> products = getQueryResult("productProjectionQueryResult.json").getResults();
+
+    @Test
+    public void testStaticJson() throws IOException {
+        final CmsPage cms = (messageKey, args) -> Optional.of(messageKey);
+        final PdpStaticData pdpStaticData = new PdpStaticData(cms);
+
+        final JsonNode expected = readJsonNodeFromResource("pdpStaticData.json");
```

Comment:
My idea was to get this JSON from the webjar directly and (in this case) access the field you want to compare. If you create your own JSON files then we have to update these JSON files every time the template is changing, which destroys the purpose of having an external library so that we can forget about frontend changes.